core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -443,9 +447,26 @@ default void setResponseTimeout(Duration responseTimeout) {
      * });
      * ...
      * }</pre>
+     *
+     * @deprecated Use {@link #whenResponseTimingOut()} or {@link #whenResponseTimedOut()}
      */
+    @Deprecated
     void setResponseTimeoutHandler(Runnable responseTimeoutHandler);
```

Comment:
Could delegate to `whenResponseTimingOut().thenRun()` and delete the old implementations?